mhs/common/mhs_common/workflow/tests/test_sync_async.py

```diff
@@ -1,66 +1,78 @@
 
-    def test_prepare_message_sync(self):
-        interaction_details = {sync_async.ASYNC_RESPONSE_EXPECTED: False}
+    def setUp(self):
+        self.persistence = MagicMock()
+        self.work_description = MagicMock()
+        self.work_description_store = MagicMock()
+        self.resync = MagicMock()
+        self.workflow = sync_async.SyncAsyncWorkflow(PARTY_ID,
+                                                     work_description_store=self.work_description_store,
+                                                     resynchroniser=self.resync)
 
-        is_async, actual_response = self.workflow.prepare_message(interaction_details,
-                                                                  sentinel.message,
-                                                                  sentinel.message_id)
+    @patch('mhs_common.state.work_description.create_new_work_description')
+    @test_utilities.async_test
+    async def test_sync_async_happy_path(self, wd_mock):
+        wd_mock.return_value = MagicMock()
+        async_workflow = MagicMock()
+        result = (200, {sync_async.MESSAGE_DATA: "Data", sync_async.CORRELATION_ID: 'cor123'})
+        async_workflow.handle_outbound_message.return_value = test_utilities.awaitable(result)
+
+        await self.workflow.handle_sync_async_outbound_message('id123', 'cor123', {},
+                                                               'payload',
+                                                               async_workflow
+                                                               )
+        wd_mock.assert_called_with(self.work_description_store,
+                                   'id123',
+                                   workflow.SYNC_ASYNC,
+                                   outbound_status=wd.MessageStatus.OUTBOUND_MESSAGE_RECEIVED)
+        async_workflow.handle_outbound_message.assert_called_once()
+
+    @patch('mhs_common.state.work_description.create_new_work_description')
+    @test_utilities.async_test
+    async def test_async_workflow_return_error_code(self, wd_mock):
+        wd_mock.return_value = MagicMock()
+        async_workflow = MagicMock()
+        result = (500, "Failed to reach spine")
+        async_workflow.handle_outbound_message.return_value = test_utilities.awaitable(result)
+
+        status, response, wdo = await self.workflow.handle_sync_async_outbound_message('id123', 'cor123', {},
+                                                                                       'payload',
+                                                                                       async_workflow
+                                                                                       )
+        self.assertEqual(status, 500)
+        self.assertEqual("Failed to reach spine", response)
 
-        self.assertFalse(is_async)
-        self.assertIs(sentinel.message, actual_response)
+    @test_utilities.async_test
+    async def test_resync_failure(self):
```

Comment:
I think it might be having an equivalent of this test case to verify that `_retrieve_async_response` updates the WDO with the expected state when the resynchroniser returns a message successfully (or the appropriate assertion could be added to the happy path test case above).

Reply:
Added it to the happy path test, this also made me aware that I wasn't checking the status code correctly either so I corrected that in the test as well